### PR TITLE
Add AAD delta outbound user and group sync

### LINKED ISSUE #323: S5+ [User Story] AAD Delta Sync
[User Story] [Epic] | Multi-party approval, Approval configuration
----------|----------
User Story Title | 
T-shirt size: S(mall), M(edium), L(arge) | M
Story Points - Fibonacci (1,2,3,5,8,13) | 5
User Story | "As an administrator of the NEXT platform implementing NEXT into my organization, I want to have the ability to import my organization's Active Directory data (users and groups) into NEXT. I need the data to be on the platform to perform RBAC duties"
Description | An Active Directory data inbound queue is needed to allow organizations to integrate their existing users and groups into the NEXT platform. Two syncs will be created. The first sync is the initial sync that pulls all the data from an organization's AAD and queues it up to be saved on the Sawtooth network. The second sync will be the delta sync. The delta sync will be used to grab any differences from the last inbound sync (could either be a initial sync or a delta sync). Both syncs will be send data to a new RethinkDB table called ad_inbound_queue.
Business Need (the why): | To allow for organizations to easily adopt the NEXT platform for their RBAC management, the platform should include an option to import users from AAD. This inbound queue will store all the records from AAD before it gets saved on the Sawtooth blockchain.
Details | AAD Initial Sync, AAD Delta Sync, New Inbound Queue table RethinkDB
Additional User Story | S4+ [User Story] LDAP AD Inbound Queue #321
Acceptance Criteria/Tests | 1) Have a new RethinkDB table called ad_inbound_queue 2) Initial sync function to grab all AAD data and add those records to ad_inbound_queue 3) Delta sync function to grab differences in AAD data from last sync and add those records to ad_inbound_queue.
Failure | Not all data records from AAD were imported to ad_inbound_queue
Definition of Done | Both syncs reflects the latest version of AD data and adds that data to the ad_inbound_queue table.                                                                                          
UX/UI Details and/or mockups | if UI change, a mockup or description of what to change, attach image/file
Applications or Systems impacted | Access to AAD test environment
Dependencies | no story dependencies
Assumptions | N/A
Performance Considerations | Syncs should be as efficient as possible as AAD could contain a large set of data.
QA Considerations | Large AAD data imports
Acrhitectural/System or Component Impacts | if we're adding something new or significantly changing architecture flag for architect review

*AAD = Azure Active Directory

### PR DESCRIPTION
**Pull request by jbobo**
Resolves #323 #434 #523

Signed-off-by: jbobo <j.ned@bobonana.me>

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -19,7 +19,7 @@ optimize-ast=no
 # can either give multiple identifiers separated by comma (,) or put this
 # option multiple times (only on the command line, not in the configuration
 # file where it should appear only once).
-disable=C0103,C0111,C0301,C0302,C0330,E1101,R0201,R0205,R0401,R0801,R0901,R0902,R0903,R0904,R0912,R0913,R0914,R0915,R1701,R1705,R1710,R1714,W503,W0612,W0613
+disable=C0103,C0111,C1801,C0301,C0302,C0330,R0123,E1101,R0201,R0205,R0401,R0801,R0901,R0902,R0903,R0904,R0912,R0913,R0914,R0915,R1701,R1705,R1710,R1714,W0511,W0612,W0613
```
Comment (adamgering) [inline setup.cfg]: new ignores must be described below

File: tests/rbac/common/role/create_role_test.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
 from rbac.common import addresser
 from rbac.common import protobuf
 from rbac.common.role.role_manager import RoleManager
+
```
Comment (adamgering) [inline tests/rbac/common/role/create_role_test.py]: don't include files unrelated to the PR

Comment (jbobo) [reply tests/rbac/common/role/create_role_test.py]: If someone submits a PR without first running black to format their touched files then they will be touched in the next PR that is properly formatted using black.

File: setup.cfg (hunk 1 above)
Comment (jbobo) [reply setup.cfg]: Those should've been removed. Rebasing.

File: tests/rbac/common/role/create_role_test.py (hunk 2 above)
Comment (adamgering) [reply tests/rbac/common/role/create_role_test.py]: still, don't include files unrelated to the PR

Comment (adamgering) [reply tests/rbac/common/role/create_role_test.py]: We'll make a separate PR soon to black non-compliant files and enforce black in Travis.

**Comment by adamgering** [comment]
How'd we go from 14 files to 3 files?

**Pull request merged**